contracts/contribute/Contributors.sol

```diff
@@ -66,6 +64,8 @@ contract Contributors {
     mapping(address => uint256) public mapMutisigActivities;
     // Mapping of whitelisted contributor agents
     mapping(address => bool) public mapContributeAgents;
+    // Mapping of whitelisted contribute managers
+    mapping(address => bool) public mapContributeManagers;
```

Comment:
Initially we need both old and new CM-s to have access, otherwise `unstake()` function will fail when called by the old one.

Reply:
Hm... I don't think we should do that. Way too messy. I will propose on a call how to do this

Reply:
`unstake()` by CM writes into contibutorsProxy to clear the msg.sender data, such as serviceId, multisig, etc. If it is not a manager then the call is reverted. So in order to unstake from old and deploy on new CM they both need to have write access for the contributors proxy.

Reply:
We don't need the old proxy at all from what I can tell. Services are forever stuck in the old CM, even with funds being retrievable.

Why not do a complete v2 deployment and have v1 just stay as is?

The benefit would be that it's easier to audit, we simply have the new system and leave Contributors.sol untouched. We move frontend to new system and have a "recovery frontend" for the current stakers.